### PR TITLE
[script][inventory-manager] Rewrite to use Lich::Util.issue_command

### PR DESCRIPTION
**Pull request by mikejcook**
Significant rewrite of inventory-manager

- Consolidated scans of items into a single method using Lich::Util.issue_command
- Corrected bug where lines over 80 characters would wrap
- Added character tag for items on the character or in their containers and 'upgrade' mechanism to address (otherwise they would never get cleaned up)
- Added home inventory (using HOME RECALL)
- Added scrolls from sort-scrolls or stack-scrolls
- Items from character, vault books, and home that are in/on containers will display the container they can be found in/on

Some examples:

```
>;inv search aura
--- Lich: inventory-manager active.
Checking Character #1:
Match 1): Aura of Tongues (2) (spell_scroll)
Found 1 match on Character #1
```

```
>;inv search lumium
--- Lich: inventory-manager active.
Checking Character #1:
Match 1): a lumium chain balaclava crafted from tempered links (worn) (character)
Match 2): a lumium chain hauberk crafted from tempered links (in container - eddy) (character)
Match 3): a lumium target shield (in container - eddy) (character)
Match 4): a lumium sallet crafted from tempered plates (in container - a sturdy purple backpack) (character)
Match 5): a lumium scale balaclava crafted from tempered scales (in container - a sturdy purple backpack) (character)
Match 6): a lumium target shield (worn) (character)
Match 7): some lumium chain gloves crafted from tempered links (worn) (character)
Match 8): a smokey glass globe suspended within a delicate lumium cage (in container - a steelsilk farmer's haversack with an onyx-hide strap) (vault)
Match 9): a smokey glass globe suspended within a delicate lumium cage (in container - a steelsilk farmer's haversack with an onyx-hide strap) (vault)
Found 9 matches on Character #1

Checking Character #2:
Match 1): a lumium ingot (in container - eddy) (character)
Match 2): a deed for a lumium ingot (in container - a heavy silk duffel bag) (character)
Match 3): a lumium ring lorica crafted from tempered links (worn) (character)
Found 3 matches on Character #2
```

```
>;inv search home
--- Lich: inventory-manager active.

Checking Character #1:
Match 1): a polished walnut floor (home)
Match 2): a tasseled cotton rug (attached to a polished walnut floor) (home)
Match 3): a silky black short-haired cat (home)
Match 4): an arched walnut door (home)
Match 5): a sturdy oak table (home)
Match 6): a polished deep green muracite ring set with a fire agate carved like a tulip blossom (attached to a sturdy oak table) (home)
Match 7): a simple stick cot (home)
Match 8): a rough-hewn oak wall (home)
Match 9): a large brick fireplace (home)
Match 10): a large oak-framed window (home)
Found 10 matches on Character #1
```

### CODE REVIEW

**Comment by MahtraDR** [comment]
I've glanced at this and it looks ok. I don't think you need to worry about the old format as long as it gets overwritten cleanly. The current script just overwrites every time you save so no fuss there. 

I'm traveling this week so limited time to look at this. I'll look in detail next week at the latest.

**Comment by mikejcook** [comment]
@MahtraDR  This is ready for a review once you're back around. If it looks okay to you, we'll need to hopefully find someone with a family vault and a storage_box to test those two items. They should work fine, I used the same messaging that was already there, but I'd feel better if they were actually tested.

**Comment by MahtraDR** [comment]
That scroll add is amazing!

I'm back from my trip. Will review asap, probably by tomorrow.

**Comment by MahtraDR** [comment]
So far, testing looks good.

Can you incorporate Shadow Servants? https://github.com/rpherbig/dr-scripts/pull/6189

And Trader shops maybe? https://github.com/rpherbig/dr-scripts/pull/6190

The only meaningful change I'd make to what you've done here is make `@inventory_ignores` a yaml setting, and store it in base.yaml instead.

**Comment by mikejcook** [comment]
@MahtraDR  Okay, all three changes are complete. I tested the trader shop changes using a standard character with shop before putting in the trader check and trader command. A regular shop works great, and so long as the trader command gets the same output (I believe it does?) that will be good.

Rubocop is throwing an error when trying to parse the script, but it doesn't indicate that there's any sort of problem. I'm not sure what it's getting hung up on.

**Comment by MahtraDR** [comment]
This looks awesome. Thank you! I'm looking into the rubocop error now.

**Comment by MahtraDR** [comment]
It's because you're not using Ruby-native syntax for two loops. I'll add comments in the code now.

File: inventory-manager.lic [hunk 1]
```diff
@@ -78,300 +92,361 @@ class InventoryManager
...
+    if stacker
+      # stacker is defined, so they're using sort-scrolls - get all ten books and parse them
+      10.times do
+        DRCI.get_item_safe?("tenth #{stacker}", stacker_container)
+        spells.concat(check_scroll_stacker(stacker))
+        DRCI.put_away_item?(stacker, stacker_container)
+      end
+    else
+      # no stacker variable, so they are using stack-scrolls. loop through their scroll stackers and parse them
+      for scroll_stacker in scroll_stackers
```
Comment (MahtraDR) [inline inventory-manager.lic:318]:
```suggestion
      scroll_stackers.each do |scroll_stacker|
```

File: inventory-manager.lic [hunk 2]
```diff
@@ -78,300 +92,361 @@ class InventoryManager
...
-                   .slice!("You rummage through a secure vault and see")
-                   .split(",")
-    item_list.each_with_index do |item, itr|
-      if itr < item_list.length - 1
-        @item_data[@active_character] << item.lstrip.concat(' (vault)')
-      else
-        item.split("and")
-            .each do |item2|
-          @item_data[@active_character] << item2.strip.concat(" (vault)").tr('.', '')
+    for surface in surfaces
```
Comment (MahtraDR) [inline inventory-manager.lic:389]:
```suggestion
    surfaces.each do |surface|
```

File: inventory-manager.lic [hunk 3]
```diff
@@ -78,300 +92,361 @@ class InventoryManager
...
+        spells.concat(check_scroll_stacker(scroll_stacker))
+        DRCI.put_away_item?(scroll_stacker, stacker_container)
       end
     end
 
-    DRC.message("Saving caravan box data for #{@active_character}!")
-    File.open(@item_db_path, 'w') { |file| file.write(@item_data.to_yaml) }
-    if using_sorter
-      start_script('sorter') unless Script.running?('sorter')
+    for spell in spells.sort
```
Comment (MahtraDR) [inline inventory-manager.lic:325]:
```suggestion
    spells.sort.each do |spell|
```

**Comment by mikejcook** [comment]
That makes sense. Have to turn off C# brain. Okay, those have been corrected and Rubocop is happy now.

**Comment by MahtraDR** [comment]
This looks awesome. Thank you! Merging now. Can you write up a fat blurb for #updates on discord?

**Comment by mikejcook** [comment]
@MahtraDR Here's the update I came up with

Update for inventory-manager.lic
The inventory-manager script has been updated with several new features to help keep tabs on inventory across characters. More specifically, several new arguments are now available for saving additional types of inventory.


Container Information
For inventory that lists containers and their contents, inventory-manager will now add the container the item can be found in for easier retrieval.

>;inv search lumium
--- Lich: inventory-manager active.

Checking CharacterName:
Match 1): a lumium ingot (in container - eddy) (character)
Match 2): a deed for a lumium ingot (in container - a heavy silk duffel bag) (character)
Match 3): a lumium ring lorica crafted from tempered links (worn) (character)
Found 3 matches on CharacterName


Spell Scrolls
;inv save scrolls
If you are using sort-scrolls or stack-scrolls, this will check your scroll stackers and save the spells and their quantities for later searching.

>;inv search aura
--- Lich: inventory-manager active.
Checking CharacterName:
Match 1): Aura of Tongues (2) (spell_scroll)
Found 1 match on CharacterName


Home
;inv save home
This command uses HOME RECALL to save all of the items in your house, including items that may be on surfaces.

>;inv search muracite
--- Lich: inventory-manager active.

Checking CharacterName:
Match 1): a polished deep green muracite ring set with a fire agate carved like a tulip blossom (attached to a sturdy oak table) (home)
Found 1 matches on CharacterName


Servant
;inv save servant
If you're a Moon Mage, this will cast Piercing Gaze on your Shadow Servant, saving the contents of its belly for later searches. You must have your Shadow Servant summoned and in the same room as you before using this command.

>;inv search servant
--- Lich: inventory-manager active.

Checking CharacterName:
Match 1): a ocarina (shadow_servant)
Match 2): a rugged gold backpack (shadow_servant)
Match 3): a purple razor (in container - a rugged gold backpack) (shadow_servant)
Match 4): a red razor (in container - a rugged gold backpack) (shadow_servant)
Found 4 matches on CharacterName


Shop
;inv save shop
This will use SHOP CUSTOMER to check the sales surfaces in your shop and save the items on/in them for later searching.

>;inv search trader_shop
--- Lich: inventory-manager active.

Checking CharacterName:
Match 1): a steel claymore for 20 platinum Kronars (a glass table) (trader_shop)
Match 2): a steel scimitar for 20 platinum Kronars (a glass table) (trader_shop)
Match 3): a lumium hauberk for 50 platinum Kronars (a steel table) (trader_shop)
Match 4): a silk duffel bag for 15 platinum Kronars (a steel table) (trader_shop)

**Comment by MahtraDR** [comment]
Thanks for that. What's your discord name for credit?

**Comment by mikejcook** [comment]
It's penguin23

**Pull request merged**